src/path.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2019  Vates SAS - ronan.abhamon@vates.fr
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "xcp-ng/generic/path.h"
+
+// =============================================================================
+
+bool xcp_path_is_abs (const char *pathname) {
+  return *pathname == '/';
+}
+
+char *xcp_path_concat (const char *pathname, const char *subpath) {
+  if (*pathname == '\0' || xcp_path_is_abs(subpath))
```

Comment:
Why `|| xcp_path_is_abs(subpath)`?

I can't do `xcp_path_concat('/my/path/', '/to/where/i.want')`?

Reply:
If you concatenate a path with an absolute path, the absolute path must be returned.
In your example `/to/where/i.want` is absolute. If you want `/my/path/to/where/i.want`, you must use a relative path like `to/where/i.want`.